salt/modules/puppet.py

```diff
@@ -246,7 +247,7 @@ def disable(message=None):
             try:
                 # Puppet chokes when no valid json is found
                 str = '{{"disabled_message":"{0}"}}'.format(message) if message is not None else '{}'
-                lockfile.write(str)
+                lockfile.write(salt.utils.stringutils.to_str(str))
```

Comment:
`str` is being shadowed here, we should be using a different name for the variable. Both lines 249 and 250 need to be updated.

Reply:
Nice catch